### DISCUSSION TITLE
Feature Request: Duplicate Meetings

### DISCUSSION THREAD

**Discussion by dougie-wougie** [code4recovery/12-step-meeting-list#1476] [Ideas]
I assume this has been asked before, but I need help locating the feature request. I want the plug-in to have a duplicate meeting feature. Since a meeting may repeat over several days, adding the first meeting, adding all the attributes, and saving it would be nice. Then, could you duplicate it for the next set of days for those meetings? 

This is a personal message to Josh. It was good meeting you at the Pacific Region, and I am ready to help with the map project. Reach out to me privately. Could you email me at webservant [at] aascv.org?

**Comment by joshreisner** [comment]
hi @dougie-wougie yes it's been asked before, but only i think with the proposed solution being to make the days be checkboxes. that approach is not good because meetings usually do have variations between instances (like big book on thursdays, for example) and the checkbox approach would not accommodate variances except via the notes field. 

your approach on the other hand has there being separate records for each meeting. i'm open to it, but we should discuss it some to align on an approach. to set expectations, we do have a fairly long list of dev priorities right now.

thanks for being willing to help with the nascent mapping effort! i'll try to write up some initial guidlines and email them to  you

**Comment by dougie-wougie** [comment]
Thank you.

It is not a road block for me. I usually make one meeting, then export the CSV file and edit in the others and reimport it.

I appreciate all that you do.

d

Sent with [Proton Mail](https://proton.me/) secure email.

**Comment by kiyote33** [comment]
FYI: the 12 Step Meeting List Enhanced Feedback plugin does what @dougie-wougie suggests. When creating a change request by selecting an existing meeting, there is an option to create a new meeting based on that existing meeting. When the "New" option is selected, some of the required fields are reset to default values for a new record (i.e. Name, etc.). The end user enters a new name, modifies other data attributes as they see fit, and create the new meeting with all the other unchanged attributes of the original meeting. 

I wonder if we might accomplish the same thing with a checkbox and some code on the TSML edit screen?

**Comment by dougie-wougie** [comment]
Thank you. I would prefer to see from the meeting listing to have a "duplicate," link which I see in similar plugs like calendars plugins. See attachment. The new meeting would change the title to "Copy of <meeting name> and it would only change the slug and other attributes necessary to make it unique. 

![CleanShot 2024-08-24 at 17 22 58@2x](https://github.com/user-attachments/assets/3be1b33c-f610-4add-a575-0c942a0d5de6)

Again, thank you for your service. 

d

**Comment by anchovie91471** [comment]
@dougie-wougie fwiw, we (area37) use this plugin to do that. 

https://wordpress.org/plugins/copy-delete-posts/

The free version doesn't copy every field though

**Reply by dougie-wougie** [reply]
I will give it a go. Thank you. 

d
